pkg/ebpf/uprobes/attacher.go

```diff
@@ -155,6 +155,19 @@ func (r *AttachRule) Validate() error {
 	return result
 }
 
+// ProcessMonitor is an interface that allows subscribing to process start and exit events
+type ProcessMonitor interface {
+	// SubscribeExec subscribes to process start events, with a callback that
+	// receives the PID of the process. Returns a function that can be called to
+	// unsubscribe from the event.
+	SubscribeExec(monitor.ProcessCallback) func()
```

Comment:
I'm not happy with having `monitor.ProcessCallback` as the type here, but it's needed to have this interface be compatible with `pkg/process/monitor.ProcessMonitor` (despite `monitor.ProcessCallback` being just `func(uint32)`, if I leave `func(uint32)` Go thinks these functions are incompatible).

In this PR we can leave it like this, personally I'd prefer to remove the type altogether and just use `func(uint32)` everywhere to avoid this problem, but we can also place this type in another common file (`pkg/util/types` maybe?)

Reply:
You can change `ProcessCallback` to be an alias to `func(uint32)`, and then remove the unneeded type here
```diff
diff --git a/pkg/process/monitor/process_monitor.go b/pkg/process/monitor/process_monitor.go
index 498d54751e..309355772d 100644
--- a/pkg/process/monitor/process_monitor.go
+++ b/pkg/process/monitor/process_monitor.go
@@ -131,7 +131,7 @@ type ProcessMonitor struct {
 }
 
 // ProcessCallback is a callback function that is called on a given pid that represents a new process.
-type ProcessCallback func(pid uint32)
+type ProcessCallback = func(pid uint32)
 
 // GetProcessMonitor create a monitor (only once) that register to netlink process events.
 //

```

Reply:
Much better, thanks a lot!